Add scraper for Discogs

dbogdanov opened the pull request:
- Uses API for search by artist + trackname
- Scrapes all found releases that have an exact simplified trackname match
- Occasionally fails with exception due to 404 error

dbogdanov committed:
Add scraper for Discogs

- Uses API for search by artist + trackname
- Scrapes all found releases that have an exact simplified trackname match
- Occasionally fails with exception due to 404 error

alastair commented:
please add `discogs_client` dependency to `requirements.txt`

alastair commented on metadb/scrapers/discogs.py:
```diff
@@ -0,0 +1,37 @@
+
+    data = None  
+    try:
+        d = discogs_client.Client('AB_Discogs/0.1', user_token="qnySAoSOYusdpbAvSvbXDnSqMyzgpATQqTLlnklk")
```
how is this different from the `DISCOGS_KEY`?

dbogdanov committed:
Remove obsolete test Discogs API key

dbogdanov committed:
Update requirements

dbogdanov committed:
Update lookup script, update discogs scraper

- Make script more generic
- Add command-line options for querying
- Add a mode to run queries based on a csv file
- Write results to files, skip scraping mbid if file exist
- Move scrapers to /release and /recording folders according to their type
- Write Discogs scraper for releases. Match releases by artist name,
  release title, and release year

dbogdanov committed:
Update discogs release scraper

- Output more details on error

dbogdanov committed:
Save empty results as well

- To avoid scraping the same MBIDs again
